### PR TITLE
bugfix: ngx.process: process.type() didn't return 'master' in master …

### PR DESCRIPTION
**Pull request by spacewander**
…process.

Previously, process.type() always returned 'single' in master process
despite whether Nginx is running in single process mode or not.

I hereby granted the copyright of the changes in this pull request
to the authors of this lua-resty-core project.

### CODE REVIEW

**Comment by spacewander** [comment]
CC @agentzh @thibaultcha

File: .travis.yml [hunk 1]
```diff
@@ -61,7 +61,7 @@ install:
   - git clone https://github.com/openresty/openresty.git ../openresty
   - git clone https://github.com/openresty/openresty-devel-utils.git
   - git clone https://github.com/simpl/ngx_devel_kit.git ../ndk-nginx-module
-  - git clone https://github.com/openresty/lua-nginx-module.git ../lua-nginx-module
+  - git clone -b fix_process_type_returns_master https://github.com/spacewander/lua-nginx-module.git ../lua-nginx-module
```
Comment (agentzh) [inline .travis.yml]: @thibaultcha Remember changing this before merging ;)

Comment (thibaultcha) [reply .travis.yml]: @agentzh Yeah I had noted that, no worries. Thanks for the reminder :)

File: lib/ngx/process.md [hunk 2]
```diff
@@ -95,14 +95,11 @@ type
...
-single
-master
-signaller
-worker
-helper
-privileged agent
-```
+* single: returned when Nginx is running in single process mode
+* signaller: returned when Nginx is running as a signaller
+* master: returned when the code is called in master process
```
Comment (agentzh) [inline lib/ngx/process.md]: Grammar: "*the* master process"

File: lib/ngx/process.md [hunk 3]
```diff
@@ -95,14 +95,11 @@ type
...
+* worker: returned when the code is called in worker process
```
Comment (agentzh) [inline lib/ngx/process.md]: Similarly, "*a* worker process"

File: lib/ngx/process.md [hunk 4]
```diff
@@ -95,14 +95,11 @@ type
...
+* privileged agent: returned when the code is called in privileged agent
```
Comment (agentzh) [inline lib/ngx/process.md]: Should be "the privileged agent process" instead.

File: lib/ngx/process.md [hunk 5]
```diff
@@ -95,14 +95,11 @@ type
...
 
-```
-single
-master
-signaller
-worker
-helper
-privileged agent
-```
+* single: returned when Nginx is running in single process mode
```
Comment (agentzh) [inline lib/ngx/process.md]: *the* single process mode.

**Comment by thibaultcha** [comment]
I will be merging this tomorrow, with maybe some slight edits in the documentation to make it look a bit nicer - thanks @spacewander!

**Comment by thibaultcha** [comment]
Merged with minor edits, thanks!

**Pull request closed without merging**